<a href="https://colab.research.google.com/github/csmotherman/CSC302HW1/blob/main/HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**github username: csmotherman**

In [ ]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
file = '/content/drive/MyDrive/DATA/olympic_medals.csv'
data = pd.read_csv(file)

*Question 2a: Find the size of the data frame. How many rows, how many columns?*

In [ ]:
data

Gender              Event Location  Year Medal                   Name  \
0         M         10000M Men      Rio  2016     G          Mohamed FARAH   
1         M         10000M Men      Rio  2016     S  Paul Kipngetich TANUI   
2         M         10000M Men      Rio  2016     B           Tamirat TOLA   
3         M         10000M Men  Beijing  2008     G        Kenenisa BEKELE   
4         M         10000M Men  Beijing  2008     S         Sileshi SIHINE   
...     ...                ...      ...   ...   ...                    ...   
2389      W  Triple Jump Women   Athens  2004     S      Hrysopiyi DEVETZI   
2390      W  Triple Jump Women   Athens  2004     B       Tatyana LEBEDEVA   
2391      W  Triple Jump Women  Atlanta  1996     G         Inessa KRAVETS   
2392      W  Triple Jump Women  Atlanta  1996     S        Inna LASOVSKAYA   
2393      W  Triple Jump Women  Atlanta  1996     B       Sarka KASPARKOVA   

     Nationality    Result  
0            GBR  25:05.17  
1            KEN  27:05.64  
2            ETH  27:06.26  
3            ETH  27:01.17  
4            ETH  27:02.77  
...          ...       ...  
2389         GRE     15.25  
2390         RUS     15.14  
2391         UKR     15.33  
2392         RUS     14.98  
2393         CZE     14.98  

[2394 rows x 8 columns]

From calling the object that we created, we can see the number of rows and columns at the bottom. **This dataset has 2394 rows and 8 columns**.

*Question 2b: Use the appropriate function to report the data types for all columns.*

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2394 entries, 0 to 2393
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Gender       2394 non-null   object
 1   Event        2394 non-null   object
 2   Location     2394 non-null   object
 3   Year         2394 non-null   int64 
 4   Medal        2394 non-null   object
 5   Name         2164 non-null   object
 6   Nationality  2394 non-null   object
 7   Result       2394 non-null   object
dtypes: int64(1), object(7)
memory usage: 149.8+ KB


*Question 2c: Find how many unique cities matches were held at.*

In [ ]:
unique_cities = data['Location'].unique()
len(unique_cities)

23

**There are 23 unique cities that the matches were held at.**

*Question 2d: Find how many medals the USA team won in total.*

In [ ]:
USA_mask = data['Nationality'] == 'USA'
USA_medals = data[USA_mask]
USA_medals

Gender             Event   Location  Year Medal             Name  \
37        M        10000M Men     London  2012     S       Galen RUPP   
51        M        10000M Men      Tokyo  1964     G      Billy MILLS   
67        M        10000M Men  Stockholm  1912     S   Lewis TEWANIMA   
70        M          100M Men        Rio  2016     S    Justin GATLIN   
74        M          100M Men    Beijing  2008     B       Walter DIX   
...     ...               ...        ...   ...   ...              ...   
2326      W  Pole Vault Women    Beijing  2008     S    Jennifer SUHR   
2328      W  Pole Vault Women     Sydney  2000     G    Stacy DRAGILA   
2331      W  Pole Vault Women     London  2012     G    Jennifer SUHR   
2337      W    Shot Put Women        Rio  2016     G  Michelle CARTER   
2358      W    Shot Put Women       Rome  1960     B    Earlene BROWN   

     Nationality    Result  
37           USA  27:30.90  
51           USA   28:24.4  
67           USA   32:06.6  
70           USA      9.89  
74           USA      9.91  
...          ...       ...  
2326         USA       4.8  
2328         USA       4.6  
2331         USA      4.75  
2337         USA     20.63  
2358         USA     16.42  

[638 rows x 8 columns]

By creating a mask where the Nationality is only "USA", we can get a list of how many medals that only the USA has won. After, we can look at the total number of rows in this new dataset. **The total number of medals that the USA won is 638.**

*Question 2e: Find the total number of medals for each Nationality.*


In [ ]:
data.groupby(by = 'Nationality')['Medal'].count()

Nationality
ALG      9
ARG      4
AUS     57
AUT      4
BAH     12
      ... 
URS    131
USA    638
VEN      2
YUG      1
ZAM      1
Name: Medal, Length: 97, dtype: int64